OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -64,8 +64,9 @@ public class ProductionQueueInfo : ITraitInfo
 		public virtual object Create(ActorInitializer init) { return new ProductionQueue(init, init.Self.Owner.PlayerActor, this); }
 	}
 
-	public class ProductionQueue : IResolveOrder, ITick, ITechTreeElement, INotifyOwnerChanged, INotifyKilled, INotifySold, ISync, INotifyTransform
+	public class ProductionQueue : IResolveOrder, INotifyCreated, ITick, ITechTreeElement, INotifyOwnerChanged, INotifyKilled, INotifySold, ISync, INotifyTransform
```

Comment:
Would it simplify things if we made this a conditional trait and replaced `Factions` with a more general condition?

Conditions would be useful to have for other use cases (e.g. a TS-style building upgrade that unlocks a new production type), so it might be better to get that out of the way now if it simplifies the other plumbing.